### PR TITLE
uri scheme & qrcode scan

### PR DESCRIPTION
**Pull request by paOol**
parse deeplinks and qrcodes with simpleledger: and bitcoincash: schemes.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-  "singleQuote": false
+  "singleQuote": false,
```
Comment (SpicyPete) [inline .prettierrc]: Revert this change.  
If we want to update `.prettierrc`, let's do it in a discrete PR where we apply the changes to the whole project.  
Adding it with another, will cause future PR's to have a lot of unrelated formatting changes.

File: screens/SendSetupScreen.js [hunk 2]
```diff
@@ -144,29 +151,37 @@ const ErrorContainer = styled(View)`
...
-      }
-    });
-  }
-  return {
-    address,
-    amount
-  };
-};
+// Only allow numbers and a single . in amount input
+const formatAmountInput = (amount: string, maxDecimals: number): string => {
```
Comment (SpicyPete) [inline screens/SendSetupScreen.js]: Remove from here, this function exists in
`utils/balance-utils`, re-use the existing one if possible

File: screens/SendSuccessScreen.js [hunk 3]
```diff
@@ -42,6 +42,7 @@ const IconImage = styled(Image)`
   height: 64;
   border-radius: 32;
   overflow: hidden;
+  border: 2px solid ${props => props.theme.bg900};
```
Comment (SpicyPete) [inline screens/SendSuccessScreen.js]: Invalid image property.  Looks better, but linter complains.  
Might need to move this to a parent div, or some other solution

**Comment by SpicyPete** [comment]
We can also remove all the images this PR is adding.  
They are leftover from the `rebase` done before

File: screens/SendSetupScreen.js [hunk 4]
```diff
@@ -144,29 +151,37 @@ const ErrorContainer = styled(View)`
...
+  const maybeZero = valid[0] && valid[0] === "." ? ["0", ...valid] : valid;
+
+  // Restrict decimals
+  const decimalIndex = maybeZero.indexOf(".");
+  const decimalAdjusted =
+    decimalIndex >= 0
+      ? maybeZero.slice(0, decimalIndex + maxDecimals + 1)
+      : maybeZero;
+
+  return decimalAdjusted.join("");
```
Comment (SpicyPete) [inline screens/SendSetupScreen.js]: Missing closing paren, so doesn't compile.  
But should use this function from the utils, so once removed here that should be fixed also

File: screens/SendSetupScreen.js [hunk 5]
```diff
@@ -295,7 +324,143 @@ const SendSetupScreen = ({
...
   };
 
+  const handleDeepLink = ({
+    address,
+    amount,
+    tokenAmount,
+    label,
+    tokenId,
+    symbol
+  }): DeepLinkParams => {
```
Comment (SpicyPete) [inline screens/SendSetupScreen.js]: DeepLinkParams could use an update.
`tokenAmount`, `tokenId`, and `symbol` are missing from it

File: screens/SendSetupScreen.js [hunk 6]
```diff
@@ -295,7 +324,143 @@ const SendSetupScreen = ({
...
+  const parseQr = (
+    qrData: string
+  ): {
+    address: string,
+    amount: ?string,
+    tokenAmount: ?object,
+    label: ?string,
+    tokenId: ?string,
+    symbol: ?string
+  } => {
```
Comment (SpicyPete) [inline screens/SendSetupScreen.js]: This type makes no sense.
`qrData` should probably be of type `DeepLinkParams`?  Or something else?
But as right now, this says the `qrData` is a string, but also an object with a bunch of fields.

File: screens/SendSuccessScreen.js (hunk 3 above)
Comment (paOol) [reply screens/SendSuccessScreen.js]: removing for now.

File: screens/SendSetupScreen.js (hunk 2 above)
Comment (paOol) [reply screens/SendSetupScreen.js]: removing

File: .prettierrc (hunk 1 above)
Comment (paOol) [reply .prettierrc]: leaving this as discussed to enforce 'defaults'.

File: screens/SendSetupScreen.js (hunk 4 above)
Comment (paOol) [reply screens/SendSetupScreen.js]: resolved from previous removal of function.

File: screens/SendSetupScreen.js (hunk 5 above)
Comment (paOol) [reply screens/SendSetupScreen.js]: those fields are needed for `prefillSLP()`.

the deeplinkparams works for both bch and slp.

Comment (SpicyPete) [reply screens/SendSetupScreen.js]: They should be added as optional parameters in the flow type in that case?

File: screens/SendSetupScreen.js (hunk 4 above)
Comment (SpicyPete) [reply screens/SendSetupScreen.js]: Then why is it being re-added in this PR...?

**Comment by paOol** [comment]
closed for a PR without previous commits.

**Pull request closed without merging**